app/services/trainees/withdraw.rb

```diff
@@ -4,14 +4,26 @@ module Trainees
   class Withdraw
     include ServicePattern
 
+    DQT_WITHDRAWN_STATE = "Withdrawn"
+
     def initialize(trainee:)
       @trainee = trainee
     end
 
     def call
+      return if withdrawn_in_dqt?
+
       Dqt::WithdrawTraineeJob.perform_later(trainee)
     end
 
+    def withdrawn_in_dqt?
+      return unless FeatureService.enabled?(:integrate_with_dqt)
+
+      return if trainee.trn.blank?
+
+      Dqt::RetrieveTeacher.call(trainee:)["initial_teacher_training"]["result"] == DQT_WITHDRAWN_STATE
```

Comment:
this might not be the way we want to go if we don't want to make another call - perhaps there is another way to check if the trainee is already withdrawn on our end and then prevent kicking off the withdraw job